src/ngraph/op/tensor_iterator.cpp

```diff
@@ -0,0 +1,128 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/tensor_iterator.hpp"
+#include "ngraph/graph_util.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+const string op::TensorIterator::type_name{"TensorIterator"};
+
+void op::TensorIterator::validate_and_infer_types()
+{
+    NODE_VALIDATION_CHECK(this,
+                          get_input_size() == m_body_parameters.size(),
+                          "Number of inputs must be the same as number of body parameters");
+
+    // The number of iterations is determined by the shortest sequence input
```

Comment:
Why we using the shortest sequence lengths? 
May be better to assume input sequences have equal lengths.

Reply:
In this version of TensorIterator, some sequence lengths were unending. Not true in the current version.
What does IE do if the sequences are different lengths?